pkg/rbd/controllerserver.go

```diff
@@ -235,6 +271,18 @@ func (cs *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 
 	rbdVol := &rbdVolume{}
 	if err := genVolFromVolID(rbdVol, volumeID, req.GetSecrets()); err != nil {
+		// If error is ErrInvalidVolID it could be a version 1.0.0 or lower volume, attempt
+		// to process it as such
+		if _, ok := err.(ErrInvalidVolID); ok {
+			if cs.MetadataStore != nil && isV1VolumeID(volumeID) {
```

Comment:
Nit: if you can detect that it's a V1 format volume ID, why not just move this block to before you attempt to lookup the volume from the omap?

Reply:
The call to `genVolFromVolID` first unpacks the `VolID` and will fail there with the error `ErrInvalidVolID`. It hence does not make any ceph/rados calls to determine the failure to decode the `VolID`. (same with the call to `getVolumeName` in the nodeserver.go file)

The further choice was to fall back to legacy rather than check legacy first. If this hampers readability (as the early failure is not obvious) let me know if it needs a reversal still.

Reply:
I didn't look at `genVolFromVolID` to see that it bails out early if it cannot decode the volume id. In that case, I think it's fine as-is.